pallets/pools/src/lib.rs

```diff
@@ -838,38 +1088,75 @@ pub mod pallet {
 					})
 					.collect::<Vec<_>>();
 
-				let tranche_values = pool
+				let tranche_supply = pool
 					.tranches
 					.iter()
 					.map(|tranche| tranche.debt.checked_add(&tranche.reserve))
 					.collect::<Option<Vec<_>>>()
 					.ok_or(Error::<T>::Overflow)?;
 
+				let seniorities = pool
+					.tranches
+					.iter()
+					.map(|tranche| tranche.seniority)
+					.collect::<Vec<_>>();
+
+				let tranche_min_risk_buffs = pool
+					.tranches
+					.iter()
+					.map(|tranche| tranche.min_risk_buffer)
+					.collect::<Vec<_>>();
+
 				let epoch_tranches: Vec<_> = orders
 					.iter()
-					.zip(&tranche_values)
+					.zip(&tranche_supply)
 					.zip(&epoch_tranche_prices)
-					.map(|(((invest, redeem), value), price)| EpochExecutionTranche {
-						value: *value,
-						price: *price,
-						invest: *invest,
-						redeem: *redeem,
-					})
+					.zip(&seniorities)
+					.zip(&tranche_min_risk_buffs)
+					.map(
+						|(((((invest, redeem), supply), price), seniority), min_risk_buffer)| {
+							EpochExecutionTranche {
+								supply: *supply,
+								price: *price,
+								invest: *invest,
+								redeem: *redeem,
+								seniority: *seniority,
+								min_risk_buffer: *min_risk_buffer,
+							}
+						},
+					)
 					.collect();
 
-				let epoch = EpochExecutionInfo {
+				let mut epoch = EpochExecutionInfo {
+					epoch: submission_period_epoch,
 					nav,
 					reserve: pool.total_reserve,
+					max_reserve: pool.max_reserve,
 					tranches: epoch_tranches,
+					best_submission: None,
+					challenge_period_end: None,
 				};
 
 				Self::deposit_event(Event::EpochClosed(pool_id, submission_period_epoch));
 
-				if Self::is_valid_solution(pool, &epoch, &full_execution_solution).is_ok() {
+				if Self::is_valid_solution(pool, &epoch, &full_execution_solution)?
+					== PoolState::Healthy
+				{
 					Self::do_execute_epoch(pool_id, pool, &epoch, &full_execution_solution)?;
 					Self::deposit_event(Event::EpochExecuted(pool_id, submission_period_epoch));
 				} else {
-					pool.submission_period_epoch = Some(submission_period_epoch);
+					// Any new submission needs to improve on the existing state (which is defined as a total fulfilment of 0%)
+					let no_execution_solution = orders
+						.iter()
+						.map(|_| TrancheSolution {
+							invest_fulfillment: Perquintill::zero(),
+							redeem_fulfillment: Perquintill::zero(),
+						})
+						.collect::<Vec<_>>();
+
+					let existing_state_solution =
+						Self::score_solution(&pool_id, &epoch, &no_execution_solution)?;
+					epoch.best_submission = Some(existing_state_solution);
```

Comment:
This approach won`t work with the current `PartialOrd` as this solution is a "healthy" one. In general, tranche-filfillments of 0 will result in a `SolutionHealthy` from the score solution. I guess this should be an extra `UnhealtyState` so that we can compare it with other unhealthy ones.